### PR TITLE
OCLOMRS-406: implement edit mapping on the right interface

### PR DESCRIPTION
**Pull request by abulojoshua1**
# JIRA TICKET NAME:
[implement edit mapping on the right interface](https://issues.openmrs.org/browse/OCLOMRS-406)

# Summary:
"edit mapping" is currently implemented in the view mappings interface. It needs to be implemented in the edit concept interface

### CODE REVIEW

File: src/components/dictionaryConcepts/containers/EditConcept.jsx [hunk 1]
```diff
@@ -77,14 +80,23 @@ export class EditConcept extends Component {
...
-    this.conceptUrl = `/${type}/${typeName}/sources/${collectionName}/concepts/${conceptId}/`;
+    this.conceptUrl = `/${type}/${typeName}/sources/${collectionName}/concepts/${conceptId}/?includeMappings=true`;
     this.props.fetchExistingConcept(this.conceptUrl);
   }
 
   componentWillUnmount() {
     this.props.clearSelections();
   }
 
+  componentWillReceiveProps(newProps) {
```
[the remark this answers is not in the document]
Comment (abulojoshua1) [reply src/components/dictionaryConcepts/containers/EditConcept.jsx]: @GodswillOnuoha, `componentWillReceiveProps` was deprecated but for applications running the older versions of React, `getDerivedStateFromProps` is not available. For our case, we will need to upgrade first to be able to use `getDerivedStateFromProps`.

File: src/components/dictionaryConcepts/containers/EditConcept.jsx [hunk 2]
```diff
@@ -104,9 +116,9 @@ export class EditConcept extends Component {
     event.preventDefault();
   }
 
-  removeNewName(event, uuid) {
+  removeNewName(event, UUID) {
```
Comment (dkayiwa) [inline src/components/dictionaryConcepts/containers/EditConcept.jsx]: What is the reason behind changing this to capital UUID?

File: src/components/dictionaryConcepts/containers/EditConcept.jsx [hunk 3]
```diff
@@ -187,6 +199,76 @@ export class EditConcept extends Component {
...
+  removeMappingRow = (event) => {
+    const { tabIndex } = event.target;
+    const { mappings } = this.state;
+    delete mappings[tabIndex];
+    this.setState({ mappings });
+  }
+
+  organizeMappings = (mappings) => {
+    const filteredMappings = mappings.map((mapping, i) => {
+      if (mapping.to_source_url === '/orgs/CIEL/sources/CIEL/') {
```
Comment (dkayiwa) [inline src/components/dictionaryConcepts/containers/EditConcept.jsx]: Why are we using the above string constant in several places? '/orgs/CIEL/sources/CIEL/'

File: src/components/dictionaryConcepts/containers/EditConcept.jsx [hunk 4]
```diff
@@ -187,6 +199,76 @@ export class EditConcept extends Component {
...
+        return {
+          id: i,
+          map_type: mapping.map_type,
+          source: 'CIEL',
```
Comment (dkayiwa) [inline src/components/dictionaryConcepts/containers/EditConcept.jsx]: Don't we have a constant for CIEL?

File: src/components/dictionaryConcepts/containers/EditConcept.jsx (hunk 3 above)
Comment (abulojoshua1) [reply src/components/dictionaryConcepts/containers/EditConcept.jsx]: @dkayiwa Let me fix that and get back to you.

File: src/components/dictionaryConcepts/containers/EditConcept.jsx (hunk 4 above)
Comment (abulojoshua1) [reply src/components/dictionaryConcepts/containers/EditConcept.jsx]: @dkayiwa Let me fix that and get back to you.

File: src/components/dictionaryConcepts/containers/EditConcept.jsx (hunk 3 above)
Comment (abulojoshua1) [reply src/components/dictionaryConcepts/containers/EditConcept.jsx]: @dkayiwa, I have already made changes.

File: src/components/dictionaryConcepts/containers/EditConcept.jsx (hunk 4 above)
Comment (abulojoshua1) [reply src/components/dictionaryConcepts/containers/EditConcept.jsx]: @dkayiwa, I have already made changes.

File: src/components/dictionaryConcepts/containers/EditConcept.jsx (hunk 2 above)
Comment (abulojoshua1) [reply src/components/dictionaryConcepts/containers/EditConcept.jsx]: It was causing conflicts, there is already a dependency with the same name that I had imported. Therefore, `uuid` was already declared in the upper scope.

Comment (dkayiwa) [reply src/components/dictionaryConcepts/containers/EditConcept.jsx]: Was that the only solution? Have you come across any convention that uses capital letters for variable names?

Comment (abulojoshua1) [reply src/components/dictionaryConcepts/containers/EditConcept.jsx]: @dkayiwa, that was not the only solution, I will just rename the variable to something better so it does not look bad.

Comment (abulojoshua1) [reply src/components/dictionaryConcepts/containers/EditConcept.jsx]: @dkayiwa, I have made changes to the variable name from `UUID` to `universalUniqueIdentifier`.

Comment (dkayiwa) [reply src/components/dictionaryConcepts/containers/EditConcept.jsx]: Which universal unique identifier is it?

**Comment by dkayiwa** [comment]
@abulojoshua1 i guess we can close this since you are no longer working on it.

**Pull request closed without merging**